src/commands/stepfunctions/helpers.ts

```diff
@@ -125,20 +125,29 @@ export const addTraceContextToStepFunctionParameters = ({Parameters}: StepType):
   }
 }
 
-export const shouldUpdateStepForTracesMerging = (step: StepType): boolean => {
+export const shouldUpdateStepForTracesMerging = (step: StepType, context: BaseContext, stepName: string): boolean => {
   // is default lambda api
   if (step.Resource === 'arn:aws:states:::lambda:invoke') {
     if (!step.Parameters) {
```

Comment:
This if statement prevents datadog-ci from crashing if parameters is missing for whatever reason. That happens before the validation you screenshotted. I think we should handle invalid configuration gracefully. Warning at this step is the right move.

Reply:
Make sense. Let's keep it then.